UI/window-basic-properties.cpp

```diff
@@ -341,6 +341,46 @@ void OBSBasicProperties::on_buttonBox_clicked(QAbstractButton *button)
 	QDialogButtonBox::ButtonRole val = buttonBox->buttonRole(button);
 
 	if (val == QDialogButtonBox::AcceptRole) {
+
+		std::string scene_name =
+			obs_source_get_name(main->GetCurrentSceneSource());
+		auto undo_redo = [this, scene_name](const std::string &data) {
```

Comment:
`this` should never be passed with objects that may be later destroyed, like the properties window. Causes a crash if you, say, create a text source, set some text, press OK, then undo the last properties change. Any instance where `this` is passed in this PR with a lambda on objects that don't persist will need to be fixed. I think what you might want to use is a QPointer object somehow or another, which is basically a helper that acts like a weak reference to any QObject.